kDrive/UI/View/Files/FileDetail/ShareLink/ShareLinkTableViewCell.swift

```diff
@@ -129,9 +134,14 @@ class ShareLinkTableViewCell: InsetTableViewCell {
                 .shareLinkRestrictedRightFolderDescriptionShort : file.isOfficeFile ? KDriveResourcesStrings.Localizable
                 .shareLinkRestrictedRightDocumentDescriptionShort : KDriveResourcesStrings.Localizable
                 .shareLinkRestrictedRightFileDescriptionShort
-            shareLinkStackView.isHidden = true
+            shareLinkStackView.isHidden = false
+            url = file.privateSharePath(host: ApiEnvironment.current.driveHost)
+            fileShareLinkSettingTitle.alpha = 0
+            fileShareLinkSettingTitle.isUserInteractionEnabled = false
             shareIconImageView.image = KDriveResourcesAsset.lock.image
         }
+
+        layoutIfNeeded()
```

Comment:
note: I moved this last, I see no good reason to do it mid-way of updating cell state.